Teacher/Teacher/SpeedGrader/MainLayout/View/DrawerContainer.swift

```diff
@@ -58,55 +63,155 @@ struct DrawerContainer<Content: View>: View {
+                    let y = height - value.predictedEndTranslation.height - minHeight
+                    let dy = maxHeight - minHeight
+                    state = (y < dy * 0.25) ? .min : (y < dy * 0.75) ? .mid : .max
+                    translation = 0
+                }
+            }
+        )
+        .accessibility(identifier: "SpeedGrader.drawerGripper")
+        .accessibilityLabel(buttonA11yText)
+    }
+
+    @ViewBuilder
+    private var openCloseButton: some View {
+        Button(action: openCloseButtonAction) {
+            openCloseButtonImage
+        }
+        .accessibilityLabel(openCloseButtonAccessibilityText)
+        .accessibilityShowsLargeContentViewer {
+            openCloseButtonImage
+            openCloseButtonAccessibilityText
+        }
+    }
+
+    private func snapDrawer(to state: DrawerState) {
+        withTransaction(DrawerState.transaction) {
+            self.state = state
+        }
+    }
+
+    private func dragIndicatorAction() {
         switch state {
-        case .min: state = .mid
-        case .mid: state = .max
-        case .max: state = .min
+        case .min: snapDrawer(to: .mid)
+        case .mid: snapDrawer(to: .max)
+        case .max: snapDrawer(to: .min)
         }
-    } }
+    }
+
+    private var buttonA11yText: Text {
+        switch state {
+        case .min: Text("Open Drawer half screen", bundle: .teacher)
+        case .mid: Text("Open Drawer full screen", bundle: .teacher)
+        case .max: Text("Close Drawer", bundle: .teacher)
+        }
+    }
+
+    @ViewBuilder
+    private var expandCollapseButtonImage: some View {
+        state == .max ? Image.exitFullScreenLine : Image.fullScreenLine
+    }
 
-    var buttonA11yText: Text {
+    private func expandCollapseButtonAction() {
         switch state {
-        case .min: return Text("Open Drawer half screen", bundle: .teacher)
-        case .mid: return Text("Open Drawer full screen", bundle: .teacher)
-        case .max: return Text("Close Drawer", bundle: .teacher)
+        case .mid: snapDrawer(to: .max)
+        default: snapDrawer(to: .mid)
         }
     }
 
+    private var expandCollapseButtonAccessibilityText: Text {
+        switch state {
+        case .min: Text("Open drawer half screen", bundle: .teacher)
+        case .mid: Text("Open drawer full screen", bundle: .teacher)
+        case .max: Text("Collapse drawer half screen", bundle: .teacher)
+        }
+    }
+
+    @ViewBuilder
+    private var openCloseButtonImage: some View {
+        Image.chevronDown
+            .rotationEffect(state != .min ? .degrees(0) : .degrees(180))
+    }
+
+    private func openCloseButtonAction() {
+        switch state {
+        case .min: snapDrawer(to: .max)
+        default: snapDrawer(to: .min)
+        }
+    }
+
+    private var openCloseButtonAccessibilityText: Text {
+        state == .min ?
+        Text("Open drawer full screen", bundle: .teacher) :
+        Text("Close drawer", bundle: .teacher)
```

Comment:
```suggestion
        state == .min
            ? Text("Open drawer full screen", bundle: .teacher)
            : Text("Close drawer", bundle: .teacher)
```
Please use this format